data_model/src/lib.rs

```diff
@@ -204,10 +204,8 @@ impl std::error::Error for ValidationError {}
 
 impl ValidationError {
     /// Construct [`ValidationError`].
-    pub fn new(reason: &str) -> Self {
-        Self {
-            reason: String::from(reason),
-        }
+    pub fn new(reason: String) -> Self {
+        Self { reason }
```

Comment:
Oh, I see there was an allocation before too, but it was hidden inside the constructor.

Reply:
Yes, it was wierd